io-engine/src/bin/io-engine-client/v0/nexus_cli.rs

```diff
@@ -336,22 +323,18 @@ async fn nexus_create(
 
 async fn nexus_create_v2(
     mut ctx: Context,
-    matches: &ArgMatches<'_>,
+    matches: &ArgMatches,
 ) -> crate::Result<()> {
     let (uuid, size, children) = nexus_create_parse(matches)?;
-    let name = matches.value_of("name").unwrap().to_string();
-    let min_cntl_id = value_t!(matches.value_of("min-cntlid"), u32)
-        .unwrap_or_else(|e| e.exit());
-    let max_cntl_id = value_t!(matches.value_of("max-cntlid"), u32)
-        .unwrap_or_else(|e| e.exit());
-    let resv_key = value_t!(matches.value_of("resv-key"), u64)
-        .unwrap_or_else(|e| e.exit());
-    let preempt_key = value_t!(matches.value_of("preempt-key"), u64)
-        .unwrap_or_else(|e| e.exit());
+    let name = matches.get_one::<String>("name").unwrap().to_string();
+    let min_cntl_id = *matches.get_one::<u32>("min-cntlid").unwrap();
+    let max_cntl_id = *matches.get_one::<u32>("max-cntlid").unwrap();
+    let resv_key = *matches.get_one::<u64>("resv-key").unwrap();
+    let preempt_key = *matches.get_one::<u64>("preempt-key").unwrap();
     let nexus_info_key = matches
-        .value_of("nexus-info-key")
-        .unwrap_or_default()
-        .to_string();
+        .get_one::<String>("nexus-info-key")
+        .unwrap()
```

Comment:
Do We need to change to unwrap_or_default() to keep same as original one ?

Reply:
It's a require parameter, it will always be set, matches the unwraps above. This is the disadvantage of using the builder pattern rather than the derive, then it would be no ambiguity.

Reply:
Oh actually it's not required because this is v0! I'll change to unwrap_or_default. 
Probably we should change to derive pattern on another PR..